iotdb-core/datanode/src/main/java/org/apache/iotdb/db/queryengine/plan/analyze/LoadTsfileAnalyzer.java

```diff
@@ -194,19 +197,49 @@ private void analyzeSingleTsFile(File tsFile) throws IOException, AuthException
       // construct tsfile resource
       final TsFileResource tsFileResource = new TsFileResource(tsFile);
       if (!tsFileResource.resourceFileExists()) {
-        if (device2TimeseriesMetadata == null) {
-          device2TimeseriesMetadata = reader.getAllTimeseriesMetadata(true);
-        }
         // it will be serialized in LoadSingleTsFileNode
         FileLoaderUtils.updateTsFileResource(device2TimeseriesMetadata, tsFileResource);
         tsFileResource.updatePlanIndexes(reader.getMinPlanIndex());
         tsFileResource.updatePlanIndexes(reader.getMaxPlanIndex());
       } else {
         tsFileResource.deserialize();
       }
+
+      final Pair<String, Long> database2WritePointCountPair =
+          getDatabase2WritePointCountPair(
+              device2TimeseriesMetadata, loadTsFileStatement.getDatabaseLevel() + 1);
+
       TimestampPrecisionUtils.checkTimestampPrecision(tsFileResource.getFileEndTime());
       tsFileResource.setStatus(TsFileResourceStatus.NORMAL);
       loadTsFileStatement.addTsFileResource(tsFileResource);
+      loadTsFileStatement.addDatabase2WritePointCountPairList(database2WritePointCountPair);
+    }
+  }
+
+  private Pair<String, Long> getDatabase2WritePointCountPair(
+      Map<String, List<TimeseriesMetadata>> device2TimeseriesMetadata,
+      int databasePrefixNodesLength)
+      throws IOException {
+    // 1. get database
+    try {
+      String device = device2TimeseriesMetadata.keySet().iterator().next();
+      final PartialPath devicePath = new PartialPath(device);
+      final String[] devicePrefixNodes = devicePath.getNodes();
+      final String[] databasePrefixNodes = new String[databasePrefixNodesLength];
+      System.arraycopy(devicePrefixNodes, 0, databasePrefixNodes, 0, databasePrefixNodesLength);
+      String database = new PartialPath(databasePrefixNodes).getFullPath();
```

Comment:
+1. In the scenario of Auth Module, frequent construct PartialPath cause serious problem. That's it is a good idea to try to use PartialPath instead of construct PartialPath every time we need.